### DISCUSSION TITLE
What is the order of execution for context and auth scope functions?

### DISCUSSION THREAD

**Discussion by aarontravass** [hayes/pothos#1183] [Q&A]
Say I have a function `extractContext` which takes the auth token as an argument and populates the context with the relevant user data. Moreover, I have an auth scope, `verifyUser` which also takes the auth token as an argument and verifies the validity of the jwt. Which function is run first?

**Comment by hayes** [comment]
I don't know when or how extractContext is used.  authScopes are run when the resolver for the first field using the authScope is run.  If you have a function that runs before the context object is created, it will run before any authScopes

**Reply by aarontravass** [reply]
> I don't know when or how extractContext is used

```ts
const yoga = createYoga<{ req: IncomingMessage }>({
  schema,
  context: (req) => extractContext(req.request.headers)
})
```

If I infer correctly, the `extractContext` function is _**always**_ run before any auth scopes?

**Reply by aarontravass** [reply]
Of course assuming the field does have an auth scope
